plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -1070,21 +1071,26 @@ private static Optional<DataCompression> getTableDataCompression(Handle handle,
 
     private static Optional<DataCompression> getTableDataCompressionWithRetries(Handle handle, JdbcTableHandle table)
     {
-        // DDL operations can take out locks against system tables causing the `getTableDataCompression` query to deadlock
-        final int maxAttemptCount = 3;
-        RetryPolicy<Optional<DataCompression>> retryPolicy = new RetryPolicy<Optional<DataCompression>>()
+        return retryOnDeadlock(() -> getTableDataCompression(handle, table), "error when getting table compression info for '%s'".formatted(table));
+    }
+
+    public static <T> T retryOnDeadlock(CheckedSupplier<T> supplier, String exceptionMessageSuffix)
```

Comment:
exceptionMessageSuffix -> exceptionMessage

Reply:
`message` makes more sense - thanks. Especially because we don't really use it in the exception - we use it to log attempts.

Reply:
Interesting. didn't think that way.